src/modinv64_impl.h

```diff
@@ -0,0 +1,419 @@
+/**********************************************************************
+ * Copyright (c) 2020 Peter Dettman                                   *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+
+#ifndef SECP256K1_MODINV64_IMPL_H
+#define SECP256K1_MODINV64_IMPL_H
+
+#include "modinv64.h"
+
+#include "util.h"
+
+static void secp256k1_modinv64_normalize_62(secp256k1_modinv64_signed62 *r, int64_t cond_negate, const secp256k1_modinv64_modinfo *modinfo) {
+    const int64_t M62 = (int64_t)(UINT64_MAX >> 2);
+    int64_t r0 = r->v[0], r1 = r->v[1], r2 = r->v[2], r3 = r->v[3], r4 = r->v[4];
+    int64_t c, cond_add;
+
+    cond_add = r4 >> 63;
+
+    c  = r0 + (modinfo->modulus.v[0] & cond_add);
+    r0 = c & M62; c >>= 62;
+    c += r1 + (modinfo->modulus.v[1] & cond_add);
+    r1 = c & M62; c >>= 62;
+    c += r2 + (modinfo->modulus.v[2] & cond_add);
+    r2 = c & M62; c >>= 62;
+    c += r3 + (modinfo->modulus.v[3] & cond_add);
+    r3 = c & M62; c >>= 62;
+    c += r4 + (modinfo->modulus.v[4] & cond_add);
+    r4 = c;
+
+    cond_add = (c >> 63) ^ cond_negate;
```

Comment:
This is wrong if input `r` is 0 (or -P) and `cond_negate` is "true", although I don't think such a call can actually happen. I have created a PR to fix it, and add _var variants as well.

Reply:
I've squashed some of the changes from that commit in here. I can't measure a performance improvement from the `_var` version (not on x86_64, arm64, or arm32), so haven't included that.